vm/src/builtins/singletons.rs

```diff
@@ -118,6 +120,22 @@ impl Representable for PyNotImplemented {
     }
 }
 
+impl Comparable for PyNone {
+    #[allow(unused_variables)]
+    fn cmp(
+        zelf: &Py<Self>,
+        other: &PyObject,
+        op: PyComparisonOp,
+        vm: &VirtualMachine,
+    ) -> PyResult<PyComparisonValue> {
```

Comment:
you didn't check `op` which holds the kind of operator (eq, ne, gt, ...)

Reply:
Yeah, I've got it as in progress as I'm using CI to test. Even with PyEnv shelled at 3.12, pytest was still picking up 3.11. Will work on it some more tonight, though I expect there'll be more to do outside of implementing the operator check

Reply:
@youknowone any reason why [this block](https://github.com/RustPython/RustPython/blob/0fab6e606379cfda346c82fa2f3960d0449b40e9/vm/src/builtins/object.rs#L105-L111) uses `PyComparison::Eq` then negates it instead of `PyComparisonOp::Ne`?

Reply:
That's a default behavior of general object. It helps a user-defined class gives peroper `!=` result when it doesn't implement `__ne__`. We can overwrite anything in subclass. Though I am not sure about it because I didn't check what's changed in recent python versions, I believe we will not need to change this part in high chance.